src/util/crypto/libcrypto/crypto_hmac_sha1.c

```diff
@@ -59,7 +59,9 @@ int sss_hmac_sha1(const unsigned char *key,
     } else {
         /* keys shorter than blocksize are zero-padded */
         memcpy(ikey, key, key_len);
-        memset(ikey + key_len, 0, HMAC_SHA1_BLOCKSIZE - key_len);
+        if (key_len != HMAC_SHA1_BLOCKSIZE) {
```

Comment:
I have them same patch in my tree just with '<' instead of '!='. I think '<' is safer because the check will still be right if by chance the if-block is moved to some other place without the 'if (key_len > HMAC_SHA1_BLOCKSIZE)' check before.

Reply:
thanks, feel free to propose your patch instead (I was just going through the e-mails today and when I made it to the Coverity notification, I coded up a patch)

Reply:
no, please go ahead, first come, first served.

Reply:
OK, I also fixed the same condition in the NSS implementation.